### PR TITLE
Add the 1Password CLI as a custom backend

### LINKED ISSUE #424: Implement secret backend for 1Password CLI
### Problem Statement

Currently there are two secret backends available: sops and vals
However none of them support the [1Password CLI](https://developer.1password.com/docs/cli/) as a secret source.

### Proposed Solution

Let's implement a new secret backend for the 1Password CLI!

### Environment

- Helm Version: v3.12.0
- Helm Secrets Version: 4.5.1
- OS: Archlinux
- Shell: bash

### Additional information

- I'm currently in the middle of implementing the secret backend myself, but after reading the contributing guidelines I decided to open an issue first.
- I'm not quite sure if I wrote the AC correctly, in case they aren't sufficient please tell me how to improve them.

### Acceptance Criteria

- [ ] As a user, I can choose between the existing sops and vals backends and the new onepassword secret backend.
- [ ] The new backend is documented in the wiki.
- [ ] Unit tests for the new onepassword secret backend have been added.

### LINKED ISSUE #434: Resolving multiple values creates invalid resources
### Current Behavior

When I have the following template:

```
apiVersion: v1
kind: Secret
metadata:
  name: test
stringData: ref+awsssm://test
```
With a value in Parameter Store in YAML format:

```
value1: secret
value2: secret
```

I get the following rendered secret:

```
apiVersion: v1
kind: Secret
metadata:
  name: test
stringData: |-
  value1: secret
  value2: secret
```

Is there a way I can extract multiple keys from a single stored parameter and render them as a map as opposed to a string?

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Environment

- Helm Version: 3.12
- Helm Secrets Version: 4.5.1
- ArgoCD Version
- OS: Linux Mint x86_64
- Shell: zsh

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jeliebig**
**What this PR does / why we need it**:

This PR adds the 1Password CLI as an additional custom backend.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #424

**Special notes for your reviewer**:

I really wanted to make the tests work as well, but for some reason `op` always thinks it is not authenticated.
Manually running the decrypt function with the test file does work as expected.
I tried a few things to make this work with BATS:
- The configuration directory for op `~/.config/op` was copied to `${HOME}/.config/op` within `setup_suite.bash`
- Additionally I tried to add a session token via an exported env var or using the `--session` parameter of `op`
- Trying to authenticate `op` within `setup_suite.bash` doesn't work, since `op` reports it is not signed in during the test run anyway

If you don't have an idea on how to potentially fix this and don't want to keep some manual test instructions for this I'll remove that commit again.

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

**Comment by bilby91** [comment]
Looking forward to testing the feature!

**Comment by jkroepke** [comment]
At the moment, I have no clue, why IT tests with coverage are failing.

**Comment by jeliebig** [comment]
> At the moment, I have no clue, why IT tests with coverage are failing.

Could that be caused by me changing things in the test libraries?

File: tests/lib/setup_suite.bash [hunk 1]
```diff
@@ -5,7 +5,13 @@ load '../lib/binaries'
 
 setup_suite() {
     {
+        if [[ "${HELM_SECRETS_BACKEND}" == "custom-"* ]]; then
```
Comment (jkroepke) [inline tests/lib/setup_suite.bash:8]: Cloud please move the if condition under line 14?

I think, in IT tests, `HELM_SECRETS_BACKEND` is undef, but its catched on line 13 here `${HELM_SECRETS_BACKEND:-"sops"}`

**Comment by jeliebig** [comment]
Hmmm, now it seems there is a problem with uploading a new report?

**Review by jkroepke — APPROVED** [review]

**Pull request merged**